modules/ztp-image-based-upgrade-concept.adoc

```diff
@@ -0,0 +1,134 @@
+
+The `ImageBasedGroupUpgrade` CR combines the `ImageBasedUpgrade` and `ClusterGroupUpgrade` APIs, sharing certain mechanisms with them.
+For example, you can define the cluster selection and rollout strategy with the `ImageBasedGroupUpgrade` API the same way that you can do with the `ClusterGroupUpgrade` API.
+
+The `ImageBasedGroupUpgrade` CR supports both a manually controlled, and a fully automated upgrade flow.
+
+[id="ztp-image-based-upgrade-supported-combinations_{context}"]
+== Supported action combinations
+
+Actions are the list of operations that the {lcao} performs in the steps of an upgrade plan.
+//Is it the LCA that performs this?
+Each `plan.actions` entry in the `ImageBasedGroupUpgrade` CR is a separate step.
+A step contains one or several actions that share the same rollout strategy.
+You can achieve more control over the rollout strategy for each action by separating actions into steps.
+These actions can be combined differently in your upgrade plan and you can append subsequent steps later.
+
+[IMPORTANT]
+====
+You cannot remove actions or steps from an ongoing plan.
+====
+//Until the IBGU is applied?
+
+The following table shows example plans for different levels of control over the rollout strategy:
+
+.Example upgrade plans
+[cols=3*, width="100%", options="header"]
+|====
+|All actions share same strategy
+|Some actions share same strategy
+|All action share same strategy
+
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep", "Upgrade", "FinalizeUpgrade"]
+    rolloutStrategy:
+      maxConcurrency: 200
+      timeout: 60
+----
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep", "Upgrade"]
+    rolloutStrategy:
+      maxConcurrency: 200
+      timeout: 60
+- actions: ["FinalizeUpgrade"]
+    rolloutStrategy:
+      maxConcurrency: 500
+      timeout: 10
+----
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep"]
+    rolloutStrategy:
+      maxConcurrency: 200
+      timeout: 60
+- actions: ["Upgrade"]
+    rolloutStrategy:
+      maxConcurrency: 200
+      timeout: 20
+- actions: ["FinalizeUpgrade"]
+    rolloutStrategy:
+      maxConcurrency: 500
+      timeout: 10
+----
+|====
+
+[IMPORTANT]
+====
+The clusters that fail one of the actions will skip the rest of the actions in the plan.
+====
+//in the plan or in the step?
+
+The {lcao} accepts the following actions:
+
+`Prep`:: Move to the `Prep` stage.
+`Upgrade`:: Move to the `Upgrade` stage.
+`FinalizeUpgrade`:: Move to the `Idle` stage to finalize a successful upgrade.
+`Rollback`:: Move to the `Rollback` stage.
+`FinalizeRollback`:: Move to the `Idle` stage to finalize a rollback.
+`AbortOnFailure`:: Move to the `Idle` stage to cancel the upgrade after a failed `Prep` or `Upgrade` action.
+`Abort`:: Move to the `Idle` stage to cancel the upgrade after the previous action.
+
+[IMPORTANT]
+====
+To avoid cancelling or rolling back an upgrade, ensure that either the `Abort` or `Rollback` action is the first step in a separate `ImageBasedGroupUpgrade` CR.
```

Comment:
```suggestion
To avoid cancelling or rolling back an upgrade on all selected clusters, ensure that either the `Abort` or `Rollback` action is the first action of a separate step added later on after assessing the overall result of the previous steps.
```